op-deployer/pkg/deployer/broadcaster/keyed.go

```diff
@@ -236,7 +235,18 @@ func padGasLimit(data []byte, gasUsed uint64, creation bool, blockGasLimit uint6
 		panic(err)
 	}
 
-	limit := uint64(float64(intrinsicGas+gasUsed) * GasPadFactor)
+	floorDataGas, err := core.FloorDataGas(data)
+	// We should never cause an overflow here.
+	if err != nil {
+		panic(err)
+	}
+
+	gas := intrinsicGas + gasUsed
+	if floorDataGas > gas {
+		gas = floorDataGas
+	}
```

Comment:
I'm not sure how `padGasLimit` is used, but added this to be on the safe side for Prague L1s

Reply:
The `floorDataGas` should be compared to the `intrinsicGas` part instead of the total, and then take the maximum of the two.

The `gasUsed` parameter refers to gas used for the call-frame, as captured during the broadcast, so that only contains EVM gas.

Reply:
I thought floorDataGas is always > intrinsicGas, but intrinsic gas + execution gas can be higher if there is execution happening. Are you sure it is wrong to compare `floorDataGas > intrinsicGas + gasUsed` here?

Reply:
intrinsic gas covers contract creation init code and calldata gas.

I think we need to check the pectra eip to verify how the floor gas is applied correctly. I thought it didn't include EVM gas.